### DISCUSSION TITLE
reducing code duplication in multiple `x-data` definitions

### DISCUSSION THREAD

**Discussion by albx79** [alpinejs/alpine#3740] [1. Help] [answered]
Hi all,

I'm new to Alpine and front-end in general (I'm mostly a back-end developer), so maybe my approach is completely wrong. But here's my problem.

I need to create an editable table with a dynamic number of rows, so I went for this approach:

```html

<table x-data="{
        template: {
           name: '',
           age: ''
        }
        items: [],
        append_item() {
            this.items.push({...this.template});
        },
        remove_last() {
            this.items.splice(this.items.length - 1, 1)
        },
        show_remove_btn(index) {
            return index == (this.items.length - 1) && index != 0;
        },
        get as_csv() { // complicated code }
}">
    <thead><tr><td>name</td><td>age</td><td></td></tr></thead>
    <tbody>
       <template x-for="(item, index) in items" :key="index">
          <tr>
              <td><input type="text" x-model="item.name"></td>
              <td><input type="text" x-model="item.age"></td>
              <td><button x-show="show_remove_btn(index)" @click="remove_last()">Remove</button></td>
          <tr>
       </template>
    </tbody>
</table>
<button @click="append_item()">Add row</button>
<button @click="code to submit the data...">Send</button>
```
(simplified; in reality there's 10-12 columns).

Now, in the same page I need to add another form with exactly the same shape and behaviour, but different data (instead of `name` and `age`, let's say that we have `make`, `model` and `year`).

Is there a way to avoid copy-pasting the whole x-data section of the first table into the second? Can I somehow inherit the method, and redefine just the template?

Thanks

**Comment by ekwoka** [comment, marked as answer]
https://alpinejs.dev/globals/alpine-data

:+1:

**Reply by albx79** [reply]
Awesome, thanks! I used the parametric form to pass the template, and it worked!
